### PR TITLE
JAMES-2506 Upgrade mock libraries (Mockito)

### PR DESCRIPTION
**Pull request by trantienduchn**

### CODE REVIEW

**Comment by trantienduchn** [comment]
> We also need to remove easymock dependency, as well as JMock

After successfully ugprade mockito. I will

**Comment by trantienduchn** [comment]
```
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] Tests run: 19, Failures: 5, Errors: 0, Skipped: 0, Time elapsed: 0.947 sec <<< FAILURE! - in org.apache.james.domainlist.lib.AbstractDomainListPrivateMethodsTest
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] containsDomainShouldReturnDetectedDomains  Time elapsed: 0.016 sec  <<< FAILURE!
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] org.junit.ComparisonFailure: expected:<[tru]e> but was:<[fals]e>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 	at org.apache.james.domainlist.lib.AbstractDomainListPrivateMethodsTest.containsDomainShouldReturnDetectedDomains(AbstractDomainListPrivateMethodsTest.java:294)
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] getDomainsShouldContainDetectedDomainsAndIpsOfAddedDomains  Time elapsed: 0.035 sec  <<< FAILURE!
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] java.lang.AssertionError: 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] Expecting:
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]  <["localhost", "93b90767126e", "added.tld", "148.25.32.2"]>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] to contain:
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]  <["detected.tld", "148.25.32.1", "added.tld", "148.25.32.2"]>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] but could not find:
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]  <["detected.tld", "148.25.32.1"]>
```

**Comment by mbaechler** [comment]
```
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] Tests run: 19, Failures: 5, Errors: 0, Skipped: 0, Time elapsed: 0.947 sec <<< FAILURE! - in org.apache.james.domainlist.lib.AbstractDomainListPrivateMethodsTest
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] containsDomainShouldReturnDetectedDomains  Time elapsed: 0.016 sec  <<< FAILURE!
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] org.junit.ComparisonFailure: expected:<[tru]e> but was:<[fals]e>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 	at org.apache.james.domainlist.lib.AbstractDomainListPrivateMethodsTest.containsDomainShouldReturnDetectedDomains(AbstractDomainListPrivateMethodsTest.java:294)
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] getDomainsShouldContainDetectedDomainsAndIpsOfAddedDomains  Time elapsed: 0.035 sec  <<< FAILURE!
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] java.lang.AssertionError: 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] Expecting:
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]  <["localhost", "93b90767126e", "added.tld", "148.25.32.2"]>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] to contain:
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]  <["detected.tld", "148.25.32.1", "added.tld", "148.25.32.2"]>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] but could not find:
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]  <["detected.tld", "148.25.32.1"]>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 	at org.apache.james.domainlist.lib.AbstractDomainListPrivateMethodsTest.getDomainsShouldContainDetectedDomainsAndIpsOfAddedDomains(AbstractDomainListPrivateMethodsTest.java:216)
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] containsDomainShouldReturnDetectedIp  Time elapsed: 0.001 sec  <<< FAILURE!
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] org.junit.ComparisonFailure: expected:<[tru]e> but was:<[fals]e>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 	at org.apache.james.domainlist.lib.AbstractDomainListPrivateMethodsTest.containsDomainShouldReturnDetectedIp(AbstractDomainListPrivateMethodsTest.java:243)
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] getDomainsShouldContainDetectedDomains  Time elapsed: 0.002 sec  <<< FAILURE!
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] java.lang.AssertionError: 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] Expecting:
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]  <[Domain : localhost, Domain : 93b90767126e]>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] to contain:
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]  <[Domain : detected.tld]>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] but could not find:
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]  <[Domain : detected.tld]>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 	at org.apache.james.domainlist.lib.AbstractDomainListPrivateMethodsTest.getDomainsShouldContainDetectedDomains(AbstractDomainListPrivateMethodsTest.java:172)
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] getDomainsShouldContainDetectedDomainsAndIps  Time elapsed: 0.002 sec  <<< FAILURE!
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] java.lang.AssertionError: 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] Expecting:
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]  <[Domain : localhost, Domain : 93b90767126e]>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] to contain:
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]  <[Domain : detected.tld, Domain : 148.25.32.1]>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] but could not find:
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]  <[Domain : detected.tld, Domain : 148.25.32.1]>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 	at org.apache.james.domainlist.lib.AbstractDomainListPrivateMethodsTest.getDomainsShouldContainDetectedDomainsAndIps(AbstractDomainListPrivateMethodsTest.java:190)
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] Results :
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] Failed tests: 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]   AbstractDomainListPrivateMethodsTest.containsDomainShouldReturnDetectedDomains:294 expected:<[tru]e> but was:<[fals]e>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]   AbstractDomainListPrivateMethodsTest.containsDomainShouldReturnDetectedIp:243 expected:<[tru]e> but was:<[fals]e>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]   AbstractDomainListPrivateMethodsTest.getDomainsShouldContainDetectedDomains:172 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] Expecting:
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]  <[Domain : localhost, Domain : 93b90767126e]>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] to contain:
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]  <[Domain : detected.tld]>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] but could not find:
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]  <[Domain : detected.tld]>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]   AbstractDomainListPrivateMethodsTest.getDomainsShouldContainDetectedDomainsAndIps:190 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] Expecting:
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]  <[Domain : localhost, Domain : 93b90767126e]>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] to contain:
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]  <[Domain : detected.tld, Domain : 148.25.32.1]>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] but could not find:
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]  <[Domain : detected.tld, Domain : 148.25.32.1]>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]   AbstractDomainListPrivateMethodsTest.getDomainsShouldContainDetectedDomainsAndIpsOfAddedDomains:216 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] Expecting:
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]  <["localhost", "93b90767126e", "added.tld", "148.25.32.2"]>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] to contain:
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]  <["detected.tld", "148.25.32.1", "added.tld", "148.25.32.2"]>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] but could not find:
[6997d9fe47d079b0347fc2c2057d77fddb3855f0]  <["detected.tld", "148.25.32.1"]>
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] Tests run: 82, Failures: 5, Errors: 0, Skipped: 0
[6997d9fe47d079b0347fc2c2057d77fddb3855f0] 
```

**Comment by mbaechler** [comment]
The previous failure is due to a change in `any` behavior : it doesn't accept "null" as argument anymore.

**Comment by mbaechler** [comment]
```
[d0db28e77d60a28a1d0e87cc5ba1b97158f27dc0] shouldThrowMessagingExceptionWhenIOExceptionReadingMessage()  Time elapsed: 0.104 sec  <<< ERROR!
[d0db28e77d60a28a1d0e87cc5ba1b97158f27dc0] org.mockito.exceptions.base.MockitoException: 
[d0db28e77d60a28a1d0e87cc5ba1b97158f27dc0] 
[d0db28e77d60a28a1d0e87cc5ba1b97158f27dc0] Checked exception is invalid for this method!
[d0db28e77d60a28a1d0e87cc5ba1b97158f27dc0] Invalid: java.io.IOException
[d0db28e77d60a28a1d0e87cc5ba1b97158f27dc0] 	at org.apache.james.transport.mailets.AddFooterTest.shouldThrowMessagingExceptionWhenIOExceptionReadingMessage(AddFooterTest.java:380)
```

**Comment by mbaechler** [comment]
I removed shouldThrowMessagingExceptionWhenIOExceptionReadingMessage because getMessage()  can't throw IOException, the test doesn't make sense

**Comment by trantienduchn** [comment]
```
[bc2c74faf8fa188202d3ffdefd580fe1d6cf9345] Results :
[bc2c74faf8fa188202d3ffdefd580fe1d6cf9345] 
[bc2c74faf8fa188202d3ffdefd580fe1d6cf9345] Tests in error: 
[bc2c74faf8fa188202d3ffdefd580fe1d6cf9345]   RecipientRewriteTableProcessorTest.processShouldSendMailToAllErrorRecipientsWhenMessagingException:291 Mockito
[bc2c74faf8fa188202d3ffdefd580fe1d6cf9345] 
[bc2c74faf8fa188202d3ffdefd580fe1d6cf9345] Tests run: 812, Failures: 0, Errors: 1, Skipped: 0

```

File: server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/RecipientRewriteTableProcessorTest.java [hunk 1]
```diff
@@ -285,31 +285,6 @@ public void processShouldSendMailToAllErrorRecipientsWhenRecipientRewriteTableEx
         assertThat(mail.getRecipients()).containsOnly(MailAddressFixture.ANY_AT_LOCAL);
     }
     
-    @SuppressWarnings("unchecked")
-    @Test
-    public void processShouldSendMailToAllErrorRecipientsWhenMessagingException() throws Exception {
```
Comment (chibenwa) [inline server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/RecipientRewriteTableProcessorTest.java]: Why?

**Comment by trantienduchn** [comment]
```
[dd8c0614a3833712197cf41ef8009cba0236e712] Tests run: 65, Failures: 0, Errors: 0, Skipped: 3, Time elapsed: 51.185 sec - in org.apache.james.queue.jms.JMSMailQueueTest
[dd8c0614a3833712197cf41ef8009cba0236e712] 
[dd8c0614a3833712197cf41ef8009cba0236e712] Results :
[dd8c0614a3833712197cf41ef8009cba0236e712] 
[dd8c0614a3833712197cf41ef8009cba0236e712] Failed tests: 
[dd8c0614a3833712197cf41ef8009cba0236e712]   AbstractMailQueueFactoryTest.destroyShouldRegisterManageableQueues:60 
[dd8c0614a3833712197cf41ef8009cba0236e712] Argument(s) are different! Wanted:
[dd8c0614a3833712197cf41ef8009cba0236e712] mBeanServer.registerMBean(
[dd8c0614a3833712197cf41ef8009cba0236e712]     <any org.apache.james.queue.api.MailQueue>,
[dd8c0614a3833712197cf41ef8009cba0236e712]     org.apache.james:type=component,name=queue,queue=queue1
[dd8c0614a3833712197cf41ef8009cba0236e712] );
[dd8c0614a3833712197cf41ef8009cba0236e712] -> at org.apache.james.queue.library.AbstractMailQueueFactoryTest.destroyShouldRegisterManageableQueues(AbstractMailQueueFactoryTest.java:60)
[dd8c0614a3833712197cf41ef8009cba0236e712] Actual invocation has different arguments:
[dd8c0614a3833712197cf41ef8009cba0236e712] mBeanServer.registerMBean(
[dd8c0614a3833712197cf41ef8009cba0236e712]     org.apache.james.queue.library.MailQueueManagement@3af0a9da,
[dd8c0614a3833712197cf41ef8009cba0236e712]     org.apache.james:type=component,name=queue,queue=queue1
[dd8c0614a3833712197cf41ef8009cba0236e712] );
[dd8c0614a3833712197cf41ef8009cba0236e712] -> at org.apache.james.queue.library.AbstractMailQueueFactory.registerMBean(AbstractMailQueueFactory.java:132)
[dd8c0614a3833712197cf41ef8009cba0236e712] 
[dd8c0614a3833712197cf41ef8009cba0236e712] 
[dd8c0614a3833712197cf41ef8009cba0236e712] Tests run: 81, Failures: 1, Errors: 0, Skipped: 3
```

**Comment by mbaechler** [comment]
```

[2d381e169d1b7332be27cdf5f7519fa125b36953] Failed tests: 
[2d381e169d1b7332be27cdf5f7519fa125b36953]   SetMessagesCreationProcessorTest.processShouldSendMailWhenRequestHasNonEmptyCreate:279 
[2d381e169d1b7332be27cdf5f7519fa125b36953] Argument(s) are different! Wanted:
[2d381e169d1b7332be27cdf5f7519fa125b36953] mailSpool.send(
[2d381e169d1b7332be27cdf5f7519fa125b36953]     <any org.apache.mailet.Mail>,
[2d381e169d1b7332be27cdf5f7519fa125b36953]     <any org.apache.james.jmap.send.MailMetadata>
[2d381e169d1b7332be27cdf5f7519fa125b36953] );
[2d381e169d1b7332be27cdf5f7519fa125b36953] -> at org.apache.james.jmap.methods.SetMessagesCreationProcessorTest.processShouldSendMailWhenRequestHasNonEmptyCreate(SetMessagesCreationProcessorTest.java:279)
[2d381e169d1b7332be27cdf5f7519fa125b36953] Actual invocation has different arguments:
[2d381e169d1b7332be27cdf5f7519fa125b36953] mailSpool.send(
[2d381e169d1b7332be27cdf5f7519fa125b36953]     null,
[2d381e169d1b7332be27cdf5f7519fa125b36953]     org.apache.james.jmap.send.MailMetadata@409eac53
[2d381e169d1b7332be27cdf5f7519fa125b36953] );
[2d381e169d1b7332be27cdf5f7519fa125b36953] -> at org.apache.james.jmap.methods.MessageSender.sendMessage(MessageSender.java:60)
[2d381e169d1b7332be27cdf5f7519fa125b36953] 
[2d381e169d1b7332be27cdf5f7519fa125b36953] 
```

**Comment by trantienduchn** [comment]
Random concurrent test fail

```
[58569954a41e2c7a8e594ffdae23bd64bbf4e08d] Results :
[58569954a41e2c7a8e594ffdae23bd64bbf4e08d] 
[58569954a41e2c7a8e594ffdae23bd64bbf4e08d] Failed tests: 
[58569954a41e2c7a8e594ffdae23bd64bbf4e08d]   DistantMailboxPathRegisterTest.unregisterMixedWithRegisterShouldWorkInAConcurrentEnvironment:328 
[58569954a41e2c7a8e594ffdae23bd64bbf4e08d] Expecting:
[58569954a41e2c7a8e594ffdae23bd64bbf4e08d]  <{namespace:user:name=116L}>
[58569954a41e2c7a8e594ffdae23bd64bbf4e08d] to contain:
[58569954a41e2c7a8e594ffdae23bd64bbf4e08d]  <[MapEntry[key=namespace:user:name, value=100L]]>
[58569954a41e2c7a8e594ffdae23bd64bbf4e08d] but could not find:
[58569954a41e2c7a8e594ffdae23bd64bbf4e08d]  <[MapEntry[key=namespace:user:name, value=100L]]>
[58569954a41e2c7a8e594ffdae23bd64bbf4e08d] 
[58569954a41e2c7a8e594ffdae23bd64bbf4e08d] 
[58569954a41e2c7a8e594ffdae23bd64bbf4e08d] Tests run: 473, Failures: 1, Errors: 0, Skipped: 0
```

test this please

**Comment by trantienduchn** [comment]
```
[58569954a41e2c7a8e594ffdae23bd64bbf4e08d] Tests in error: 
[58569954a41e2c7a8e594ffdae23bd64bbf4e08d]   ForwardRoutesTest$ExceptionHandling.deleteShouldReturnErrorWhenErrorMappingExceptionIsThrown:679 Mockito
[58569954a41e2c7a8e594ffdae23bd64bbf4e08d]   ForwardRoutesTest$ExceptionHandling.getAllShouldReturnErrorWhenErrorMappingExceptionIsThrown:643 Mockito
[58569954a41e2c7a8e594ffdae23bd64bbf4e08d]   ForwardRoutesTest$ExceptionHandling.getShouldReturnErrorWhenErrorMappingExceptionIsThrown:715 Mockito
[58569954a41e2c7a8e594ffdae23bd64bbf4e08d]   ForwardRoutesTest$ExceptionHandling.putShouldReturnErrorWhenErrorMappingExceptionIsThrown:607 Mockito
[58569954a41e2c7a8e594ffdae23bd64bbf4e08d] 
[58569954a41e2c7a8e594ffdae23bd64bbf4e08d] Tests run: 167, Failures: 0, Errors: 4, Skipped: 0
```

**Comment by trantienduchn** [comment]
things seem are normal but failure at the end of workflow
test this please

File: protocols/imap/src/test/java/org/apache/james/imap/decode/parser/FetchCommandParserPartialFetchTest.java [hunk 2]
```diff
@@ -75,18 +66,12 @@ public void testShouldParseNonZeroAndLength() throws Exception {
         check("1 (BODY[]<20.12342348>)\r\n", ranges, false, data, "A01");
     }
 
-    @Test
+    @Test(expected = DecodingException.class)
```
Comment (chibenwa) [inline protocols/imap/src/test/java/org/apache/james/imap/decode/parser/FetchCommandParserPartialFetchTest.java]: Prefer use `assertThatThrownBy`please

File: protocols/imap/src/test/java/org/apache/james/imap/encode/FetchResponseEncoderEnvelopeTest.java [hunk 3]
```diff
@@ -122,33 +119,26 @@ public void setUp() throws Exception {
...
-        final Address address = context.mock(Address.class, name + host);
-        context.checking(new Expectations() {{
-                    oneOf(address).getPersonalName();will(returnValue(name));
-                    oneOf(address).getAtDomainList();will(returnValue(domainList));
-                    oneOf(address).getMailboxName();will(returnValue(mailbox));
-                    oneOf(address).getHostName();will(returnValue(host));
-                }
-            }
-        );
+        final Address address = mock(Address.class);
```
Comment (chibenwa) [inline protocols/imap/src/test/java/org/apache/james/imap/encode/FetchResponseEncoderEnvelopeTest.java]: We don't need final variables and final arguments anymore.

File: protocols/imap/src/test/java/org/apache/james/imap/encode/FetchResponseEncoderNoExtensionsTest.java [hunk 4]
```diff
@@ -109,20 +104,18 @@ public void testShouldNotAddExtensionsWithEncodingBodyStructure() throws Excepti
...
-                    allowing(stubStructure).getOctets();will(returnValue(octets));
-                    allowing(stubStructure).getLines();will(returnValue(lines));
-                    allowing(stubStructure).getParameters(); will(returnValue(parameterList));
-                    allowing(stubStructure).getEncoding(); will(returnValue("7BIT"));
-                    ignoring(stubStructure);
-                }
-            }
-        );
+
+        final long octets = 2279L;
```
Comment (chibenwa) [inline protocols/imap/src/test/java/org/apache/james/imap/encode/FetchResponseEncoderNoExtensionsTest.java]: final can be removed

File: protocols/imap/src/test/java/org/apache/james/imap/processor/GetQuotaRootProcessorTest.java [hunk 5]
```diff
@@ -71,122 +74,78 @@
...
-                oneOf(mockedResponder).respond(with(new StatusResponseTypeMatcher(StatusResponse.Type.BAD)));
-            }
-        });
+        verify(mockedMailboxManager).startProcessingRequest(mailboxSession);
+        verify(mockedMailboxManager).endProcessingRequest(mailboxSession);
 
-        testee.doProcess(getQuotaRootRequest, mockedResponder, mockedImapSession);
+        verify(mockedResponder, atLeastOnce()).respond(argThat(response ->
+            (response instanceof StatusResponse)
+            && ((StatusResponse) response).getServerResponseType().equals(StatusResponse.Type.BAD)));
```
Comment (chibenwa) [inline protocols/imap/src/test/java/org/apache/james/imap/processor/GetQuotaRootProcessorTest.java]: Why can't we keep equality here?

File: protocols/imap/src/test/java/org/apache/james/imap/processor/GetQuotaRootProcessorTest.java [hunk 6]
```diff
@@ -71,122 +74,78 @@
...
-                oneOf(mockedResponder).respond(with(new StatusResponseTypeMatcher(StatusResponse.Type.NO)));
-            }
-        });
+        verify(mockedMailboxManager).startProcessingRequest(mailboxSession);
+        verify(mockedMailboxManager).endProcessingRequest(mailboxSession);
 
-        testee.doProcess(getQuotaRootRequest, mockedResponder, mockedImapSession);
+        verify(mockedResponder, atLeastOnce()).respond(argThat(response ->
+            (response instanceof StatusResponse)
+            && ((StatusResponse) response).getServerResponseType().equals(StatusResponse.Type.NO)));
```
Comment (chibenwa) [inline protocols/imap/src/test/java/org/apache/james/imap/processor/GetQuotaRootProcessorTest.java]: Idem.

**Comment by trantienduchn** [comment]
Lucene search fails
```
[01d7d2118af4621ac4b59851910d1109c57b9309] Results :
[01d7d2118af4621ac4b59851910d1109c57b9309] 
[01d7d2118af4621ac4b59851910d1109c57b9309] Tests in error: 
[01d7d2118af4621ac4b59851910d1109c57b9309]   Search.testSearchAtomsITALY:63 » InvalidServerResponse 
[01d7d2118af4621ac4b59851910d1109c57b9309] Location: /org/apache/...
[01d7d2118af4621ac4b59851910d1109c57b9309]   Search.testSearchAtomsKOREA:71 » InvalidServerResponse 
[01d7d2118af4621ac4b59851910d1109c57b9309] Location: /org/apache/...
[01d7d2118af4621ac4b59851910d1109c57b9309]   Search.testSearchAtomsUS:55 » InvalidServerResponse 
[01d7d2118af4621ac4b59851910d1109c57b9309] Location: /org/apache/jam...
[01d7d2118af4621ac4b59851910d1109c57b9309]   Search.testSearchCombinationsITALY:85 » InvalidServerResponse 
[01d7d2118af4621ac4b59851910d1109c57b9309] Location: /org/...
[01d7d2118af4621ac4b59851910d1109c57b9309]   Search.testSearchCombinationsKOREA:92 » InvalidServerResponse 
[01d7d2118af4621ac4b59851910d1109c57b9309] Location: /org/...
[01d7d2118af4621ac4b59851910d1109c57b9309]   Search.testSearchCombinationsUS:78 » InvalidServerResponse 
[01d7d2118af4621ac4b59851910d1109c57b9309] Location: /org/apa...
[01d7d2118af4621ac4b59851910d1109c57b9309]   SelectedState.testSearchITALY:130 » InvalidServerResponse 
[01d7d2118af4621ac4b59851910d1109c57b9309] Location: /org/apac...
[01d7d2118af4621ac4b59851910d1109c57b9309]   SelectedState.testSearchKOREA:193 » InvalidServerResponse 
[01d7d2118af4621ac4b59851910d1109c57b9309] Location: /org/apac...
[01d7d2118af4621ac4b59851910d1109c57b9309]   SelectedState.testSearchUS:67 » InvalidServerResponse 
[01d7d2118af4621ac4b59851910d1109c57b9309] Location: /org/apache/j...
[01d7d2118af4621ac4b59851910d1109c57b9309]   SelectedState.testUidITALY:172 » InvalidServerResponse 
[01d7d2118af4621ac4b59851910d1109c57b9309] Location: /org/apache/...
[01d7d2118af4621ac4b59851910d1109c57b9309]   SelectedState.testUidKOREA:235 » InvalidServerResponse 
[01d7d2118af4621ac4b59851910d1109c57b9309] Location: /org/apache/...
[01d7d2118af4621ac4b59851910d1109c57b9309]   SelectedState.testUidUS:109 » InvalidServerResponse 
[01d7d2118af4621ac4b59851910d1109c57b9309] Location: /org/apache/jam...
[01d7d2118af4621ac4b59851910d1109c57b9309]   UidSearchOnIndex.testSearchAtomsUS:52 » InvalidServerResponse 
[01d7d2118af4621ac4b59851910d1109c57b9309] Location: /org/...
[01d7d2118af4621ac4b59851910d1109c57b9309]   UidSearch.testSearchAtomsITALY:59 » InvalidServerResponse 
[01d7d2118af4621ac4b59851910d1109c57b9309] Location: /org/apac...
[01d7d2118af4621ac4b59851910d1109c57b9309]   UidSearch.testSearchAtomsKOREA:66 » InvalidServerResponse 
[01d7d2118af4621ac4b59851910d1109c57b9309] Location: /org/apac...
[01d7d2118af4621ac4b59851910d1109c57b9309]   UidSearch.testSearchAtomsUS:52 » InvalidServerResponse 
[01d7d2118af4621ac4b59851910d1109c57b9309] Location: /org/apache/...
[01d7d2118af4621ac4b59851910d1109c57b9309]   UidSearch.testSearchCombinationsITALY:80 » InvalidServerResponse 
[01d7d2118af4621ac4b59851910d1109c57b9309] Location: /o...
[01d7d2118af4621ac4b59851910d1109c57b9309]   UidSearch.testSearchCombinationsKOREA:87 » InvalidServerResponse 
[01d7d2118af4621ac4b59851910d1109c57b9309] Location: /o...
[01d7d2118af4621ac4b59851910d1109c57b9309]   UidSearch.testSearchCombinationsUS:73 » InvalidServerResponse 
[01d7d2118af4621ac4b59851910d1109c57b9309] Location: /org/...
[01d7d2118af4621ac4b59851910d1109c57b9309] 
[01d7d2118af4621ac4b59851910d1109c57b9309] Tests run: 315, Failures: 0, Errors: 19, Skipped: 7
```

**Comment by chibenwa** [comment]
Well, lucene is a **search index** thus we should only play search index tests on it, just like ElasticSearch. If we play regular search test suite, it will fail, for sure!

**Comment by mbaechler** [comment]
please keep only mockito upgrade in this PR and have it merged.
Open another PR with jmock/easymock removal.

**Comment by trantienduchn** [comment]
okay

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**